pumps/resurface.go

```diff
@@ -0,0 +1,217 @@
+package pumps
+
+import (
+	"context"
+	"encoding/base64"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/TykTechnologies/tyk-pump/analytics"
+	"github.com/mitchellh/mapstructure"
+	"github.com/pkg/errors"
+	"github.com/resurfaceio/logger-go/v3"
+)
+
+type ResurfacePump struct {
+	logger *logger.HttpLogger
+	config *ResurfacePumpConfig
+	CommonPumpConfig
+}
+
+type ResurfacePumpConfig struct {
+	EnvPrefix string `mapstructure:"meta_env_prefix"`
+	Url       string `mapstructure:"capture_url"`
+	Rules     string
+	Queue     []string
+}
+
+const (
+	resurfacePrefix     = "resurface-pump"
+	resurfacePumpName   = "Resurface Pump"
+	resurfaceDefaultEnv = PUMPS_ENV_PREFIX + "_RESURFACEIO" + PUMPS_ENV_META_PREFIX
+)
+
+func (rp *ResurfacePump) New() Pump {
+	newPump := ResurfacePump{}
+	return &newPump
+}
+
+func (rp *ResurfacePump) GetName() string {
+	return resurfacePumpName
+}
+
+func (rp *ResurfacePump) GetEnvPrefix() string {
+	return rp.config.EnvPrefix
+}
+
+func (rp *ResurfacePump) Init(config interface{}) error {
+	rp.config = &ResurfacePumpConfig{}
+	rp.log = log.WithField("prefix", resurfacePrefix)
+
+	err := mapstructure.Decode(config, &rp.config)
+	if err != nil {
+		rp.log.Fatal("Failed to decode configuration: ", err)
```

Comment:
I don't think we should `Fatal` here. Maybe just a debug log can be enough

Reply:
Thank you, you are absolutely right. I just modified this.